src/base/DelegateInputLabelMixin.js

```diff
@@ -31,18 +32,60 @@ export default function DelegateInputLabelMixin(Base) {
       this[setState]({ ariaLabel });
     }
 
+    // Forward ARIA labelledby as an aria-label to the input element.
+    get ariaLabelledby() {
+      return this[state].ariaLabelledby;
+    }
+    set ariaLabelledby(ariaLabelledby) {
+      if (this.getRootNode() !== null) {
+        // @ts-ignore
+        let labelNode = this.getRootNode().querySelector(
+          `#${ariaLabelledby}`
+        );
+
+        labelNode.setAttribute('aria-hidden', true);
+        const ariaLabel = labelNode.innerText;
+        this[setState]({ ariaLabel, ariaLabelledby });
```

Comment:
One side-effect of using the `ariaLabel` property to hold the label we're constructing is that the `ariaLabel` property will return the label — even if the dev is setting `aria-labelledby`.

      <label for="input-0">Email</label>
      <elix-input id="input-0"></elix-input>

If the dev asks for the `ariaLabel` property of the `elix-input`, it will return "Email", even though the dev hasn't set `ariaLabel`.

I think we probably should have a separate piece of state called `label` or something that's entirely internal. Setting ariaLabel and ariaLabelledby would have the side effect of setting that `label` state, but a dev wouldn't be able to get the value of that state directly. I can explain how to do that the next time we talk.